srcalc/src/main/java/gov/va/med/srcalc/web/controller/CalculationController.java

```diff
@@ -44,6 +44,39 @@ public ModelAndView startNewCalculation(
         mav.addObject("specialties", fCalculationService.getValidSpecialties());
         return mav;
     }
+
+    @RequestMapping(value = "/newCalc", method = RequestMethod.GET)
+    public ModelAndView startNewCalculation(
+            @RequestParam(value = "patientDfn") final int patientDfn,
+            final HttpSession session)
+    {
+        // If there is a calculation already in the session, ask the user
+        // if they wish to override the in-progress calculation.
+        try
+        {
+            final CalculationSession calcSession = SrcalcSession.getCalculationSession(session);
+            final Calculation calc = calcSession.getCalculation();
+            final ModelAndView mav = new ModelAndView("redirect:/confirmNewCalc");
```

Comment:
Why not just show the view here instead of redirecting? What is the advantage of redirecting?